Client/Frontend/Menu/MenuConfiguration.swift

```diff
@@ -4,146 +4,206 @@
     }
 
     func menuFont() -> UIFont {
         return UIFont.systemFontOfSize(11)
     }
-}
-
-// MARK: Static helper access function
 
-extension MenuConfiguration {
+    func menuIcon() -> UIImage? {
+        return isPrivateMode ? UIImage(named:"bottomNav-menu-pbm") : UIImage(named:"bottomNav-menu")
+    }
 
-    static func menuIconForMode(isPrivate isPrivate: Bool = false) -> UIImage? {
-        return isPrivate ? UIImage(named:"bottomNav-menu-pbm") : UIImage(named:"bottomNav-menu")
+    func shadowColor() -> UIColor {
+        return isPrivateMode ? UIColor.darkGrayColor() : UIColor.lightGrayColor()
     }
 
-    static func menuConfigurationForLocation(location: MenuLocation) -> MenuConfiguration {
-        return MenuConfiguration(location: location, menuItems: menuItemsForLocation(location), toolbarItems: menuToolbarItemsForLocation(location), numberOfItemsInRow: numberOfMenuItemsPerRowForLocation(location))
+    func selectedItemTintColor() -> UIColor {
+        return UIConstants.MenuSelectedItemTintColor
     }
 
-    private static func numberOfMenuItemsPerRowForLocation(location: MenuLocation) -> Int {
-        switch location {
+    private func numberOfMenuItemsPerRowForAppState(appState: AppState) -> Int {
+        switch appState {
         case .TabTray:
-           return 4
+            return 4
         default:
             return 3
         }
     }
 
     // the items should be added to the array according to desired display order
-    private static func menuItemsForLocation(location: MenuLocation) -> [MenuItem] {
-        let menuItems: [MenuItem]
-        switch location {
-        case .Browser:
-            // TODO: filter out menu items that are not to be displayed given the current app state
-            // (i.e. whether the current browser URL is bookmarked or not)
-            menuItems = [FindInPageMenuItem, RequestDesktopMenuItem, RequestMobileMenuItem, SettingsMenuItem, NewTabMenuItem, NewPrivateTabMenuItem, AddBookmarkMenuItem, RemoveBookmarkMenuItem]
+    private func menuItemsForAppState(appState: AppState) -> [MenuItem] {
+        var menuItems = [MenuItem]()
+        switch appState {
+        case .Tab(let tabState):
+            menuItems.append(FirefoxMenuConfiguration.FindInPageMenuItem)
+            if #available(iOS 9, *) {
+                menuItems.append(tabState.desktopSite ? FirefoxMenuConfiguration.RequestMobileMenuItem : FirefoxMenuConfiguration.RequestDesktopMenuItem)
+            }
+            menuItems.append(FirefoxMenuConfiguration.SettingsMenuItem)
+            menuItems.append(FirefoxMenuConfiguration.NewTabMenuItem)
+            if #available(iOS 9, *) {
+                menuItems.append(FirefoxMenuConfiguration.NewPrivateTabMenuItem)
+            }
+            menuItems.append(tabState.isBookmarked ? FirefoxMenuConfiguration.RemoveBookmarkMenuItem : FirefoxMenuConfiguration.AddBookmarkMenuItem)
```

Comment:
I'm not wild about having the configuration determine if an item should be included. But OK.